Update dependencies: RN 0.63.4, TS 4.1.3, etc.

jerome-benoit opened the pull request:
And do: 
+ the typing to silence tsc
+ proper indentation

Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

LucasBrazi06 commented on src/components/header/HeaderComponent.tsx:
```diff
@@ -11,11 +11,11 @@ import computeStyleSheet from './HeaderComponentStyles';
   title: string;
   subTitle?: string;
-  leftAction?: () => void;
+  leftAction?: () => boolean;
```
Why changing it to a boolean?

LucasBrazi06 commented on src/components/header/HeaderComponent.tsx:
```diff
@@ -11,11 +11,11 @@ import computeStyleSheet from './HeaderComponentStyles';
   leftActionIconType?: IconType;
   hideHomeAction?: boolean;
-  rightAction?: () => void;
+  rightAction?: () => boolean;
```
Why changing it to a boolean?

jerome-benoit replied:
It's the type expected by react-native BackHandler.addEventListener() for that callback. 
If you run `npm run tsc`, you have that type mismatch in the list.

LucasBrazi06 replied:
Okay

LucasBrazi06 reviewed (commented):
Big refactoring of tenant management is in progress now, so this PR is put on hold.

LucasBrazi06 commented on src/components/transaction/history/TransactionHistoryComponent.tsx:
```diff
@@ -56,9 +56,9 @@ export default class TransactionHistoryComponent extends React.Component<Props,
-                params: { transactionID: transaction.id },
-                key: `${Utils.randomNumber()}`
-              }
+              params: { transactionID: transaction.id },
```
Not indented properly!

jerome-benoit replied:
It's object literals passed as arguments of a method or function. 
And it's properly indented: 
+ each arguments have the same indentation level; 
+ each keys in objects literal passed as arguments have the same indentation level. 

Before the indentation level was considering only one object literal as the argument and method arguments as object literal keys, which is not correct. The re-indentation in that PR have been made automatically by code editor and linter. 

All the indentation fixlet done in that PR correct it.

LucasBrazi06 replied:
I see that, 'param' should be indented:
![image](https://user-images.githubusercontent.com/25266945/103150820-0a562f00-4778-11eb-8a03-25aa09127d94.png)

jerome-benoit replied:
> I see that, 'param' should be indented:
> ![image](https://user-images.githubusercontent.com/25266945/103150820-0a562f00-4778-11eb-8a03-25aa09127d94.png)

After adding a new line separator for each arguments, the automatic indentation now works.

LucasBrazi06 commented:
@jerome-benoit Stop mixing formatting and upgrading of modules!!!
The formatting is wrong here and I cannot merge the PR!
Always doing the same mistake!!!!

jerome-benoit commented:
> @jerome-benoit Stop mixing formatting and upgrading of modules!!!
> The formatting is wrong here and I cannot merge the PR!
> Always doing the same mistake!!!!

It's just a mix of non functional changes like import cleanups, typing, indentation, etc. with dependencies upgrade which is the only functional change made that need more testing. It's just a convenient way to coin them in the code flow without multiplying PRs and make them easy to review: two files only for dependencies upgrade, the rest can only be non functional changes and that way they are not mixed in PRs with important functional changes. So if there's issue, it can only be because of the dependency upgrade.   
That PR still have issue regarding connectors status showed but I'm not sure about the root cause yet. I put then that PR in the 'In Progress' pipeline. 

And the indentation is correct in that PR like explained in another comment.

jerome-benoit commented:
> > @jerome-benoit Stop mixing formatting and upgrading of modules!!!
> > The formatting is wrong here and I cannot merge the PR!
> > Always doing the same mistake!!!!
> 

> That PR still have issue regarding connectors status showed but I'm not sure about the root cause yet. I put then that PR in the 'In Progress' pipeline.

I do not understand why on offline charging station, the connector status is incorrect. The re-indentation and incorrect switch case construct fix could trigger that issue . 

> And the indentation is correct in that PR like explained in another comment ;).

Automatic indentation should now work properly with separating arguments in the call.

jerome-benoit commented:
> > That PR still have issue regarding connectors status showed but I'm not sure about the root cause yet. I put then that PR in the 'In Progress' pipeline.
> 
> I do not understand why on offline charging stations, the connector status is incorrect. The re-indentation and incorrect switch case construct fix in `getConnectorStyles()` should not trigger that issue .

The ev-mobile application just displays the connector status the REST API return. On the dashboard, the connector status is forced to 'Unavailable' if the charging station is considered as offline.
We can: 
+ do the same on the mobile app;
+ ensure the connectors status are set to 'Unavailable' when a charging station is flagged as offline server side. 
I prefer the later. 

That issue is then unrelated to that PR.

LucasBrazi06 commented:
> > @jerome-benoit Stop mixing formatting and upgrading of modules!!!
> > The formatting is wrong here and I cannot merge the PR!
> > Always doing the same mistake!!!!
> 
> It's just a mix of non functional changes like import cleanups, typing, indentation, etc. with dependencies upgrade which is the only functional change made that need more testing. It's just a convenient way to coin them in the code flow without multiplying PRs and make them easy to review: two files only for dependencies upgrade, the rest can only be non functional changes and that way they are not mixed in PRs with important functional changes. So if there's issue, it can only be because of the dependencies upgrade.
> That PR still have issue regarding connectors status showed but I'm not sure about the root cause yet. I put then that PR in the 'In Progress' pipeline.
> 
> And the indentation is correct in that PR like explained in another comment ;).

Same comment :-)
I cannot read your updates as there are too many changes.
Next time create a new PR for this purpose or I'll close it.

LucasBrazi06 commented:
> The ev-mobile application just displays the connector status the REST API return. On the dashboard, the connector status is forced to 'Unavailable' if the charging station is considered as offline.

We have to do it at UI level to avoid processing time at backend level.
Moreover, you may want also to know the latest status of the connectors before it went offline.

LucasBrazi06 commented:
> > The ev-mobile application just displays the connector status the REST API return. On the dashboard, the connector status is forced to 'Unavailable' if the charging station is considered as offline.
> 
> We have to do it at UI level to avoid processing time at backend level.
> Moreover, you may want also to know the latest status of the connectors before it went offline.

I just did it in V1.3.33

jerome-benoit commented:
> > > @jerome-benoit Stop mixing formatting and upgrading of modules!!!
> > > The formatting is wrong here and I cannot merge the PR!
> > > Always doing the same mistake!!!!

> > It's just a mix of non functional changes like import cleanups, typing, indentation, etc. with dependencies upgrade which is the only functional change made that need more testing. It's just a convenient way to coin them in the code flow without multiplying PRs and make them easy to review: two files only for dependencies upgrade, the rest can only be non functional changes and that way they are not mixed in PRs with important functional changes. So if there's issue, it can only be because of the dependencies upgrade.
> > That PR still have issue regarding connectors status showed but I'm not sure about the root cause yet. I put then that PR in the 'In Progress' pipeline.
> > And the indentation is correct in that PR like explained in another comment ;).
> 
> Same comment :-)
> I cannot read your updates as there are too many changes.
> Next time create a new PR for this purpose or I'll close it.

And then you will complain about having too much PRs not attached to an issue: 
+ one for typing
+ one for code style
+ one for dependencies update
+ one for logic misconstruct 
+ ... 

;)

Sometimes rules need some small violation to fit in. 
Usually there's less non functional unrelated changes to review but that PR is opened for more than a month: more important changes were expected to go in before its merge. So the size of the unrelated changes have increased over time during the wait. 

In any case, I've addressed the comments done in the review, tested it (need `npm run android:clean`, `npm run pod:install`, etc.). It contains some genuine fixes related to the RN style typing, logic misconstruct or security.

LucasBrazi06 commented:
> And then you will complain about having too much PRs not attached to an issue:

You almost never do that anyway!

> * one for typing
> * one for code style
> * one for dependencies update
> * one for logic misconstruct
> * ...

You can group some of these topics together

> Sometimes rules need some small violation to fit in.

It's more a rule than an exception with you ;-)

> Usually there's less non functional unrelated changes to review but that PR is opened for more than a month: more
> important changes were expected to go in before its merge. So the size of the unrelated changes have increased over time
> during the wait.


That's normal, I do not merge when you update RN version that can impact current dev before an imminent prod release you are not aware of. But I can merge code style changes...

jerome-benoit commented:
> In any case, I've addressed the comments done in the review, tested it (needs `npm run android:clean`, `npm run pod:install`, etc.). It contains some genuine fixes related to the RN style typing and composition, logic misconstruct or security.

Conflicts resolved (again). 
The dependencies upgrade are either against the correction or patch in their SemVer, except for one for security reason. So it's a safe upgrade path and the update that might contains breaking changes have been tested to work as intended.
